### DISCUSSION TITLE
Measure or determine lux in image/camera

### DISCUSSION THREAD

**Discussion by mortalone** [blakeblackshear/frigate#8287] [Ideas]
Hi. i think it would be realy cool if it was possible to measure the lux in the picture/frame

this could come in handy when making automations. lets say i only want the light to turn on when the lux is below a certain point (i know i can do this with other devices or generic sensors that are based on location and sun location. but they do not take in account if it is cloudy. and thus darker than usual at another time of day. i hope it makes sense.).

I know that this cannot be a fixed set of values. and has to be tuned for every camera. but i still feel like the feature would be very handy.

I emagine if possible that you are able to sellect a certain area like the masking tool. where frigate is able to measure the color/brightness or whatever you clever dudes do with your magic. and then via configuration set a max/min value or just get the average value of that area. and use that as a "lux" or at least a pont for reference. AND then expose this value to MQTT

I hope it is clear what im thinking.

And i have to bow for the awesome work you guys do on this software. it is simply awesome. and i love it.

**Comment by hagak** [comment]
Problem here is I think most ip cameras use auto exposure control without reporting the exposure.  In order for this to work the camera would need to report the exposure setting it is currently using and I do not believe many if any do.  Without that you really could not even tune per camera.  If you can disable auto-exposure that could potential work but then you would have a camera that is not useful at all at doing camera work since the images would be either way too dark or way to bright and rarely just right.

**Reply by mortalone** [reply]
i see. dang it :( would be nice

**Reply by Icantpickaname** [reply]
This as a feature would be brilliant. I gathered it would either be a major chore to implement or impossible. Thanks for the clarification @hagak

**Comment by x-real-ip** [comment]
Hi, I made is-it-dark, a small API that checks if an image is dark or bright. It also returns the calculated brightness value and detects if the camera is in infrared mode. It’s useful for automations or analyzing camera images. The service can run containerised.

https://github.com/x-real-ip/is-it-dark
